oap-server/server-receiver-plugin/skywalking-sharing-server-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/sharing/server/SharingServerModuleProvider.java

```diff
@@ -67,26 +68,38 @@ public ModuleConfig createConfigBeanIfAbsent() {
 
     @Override
     public void prepare() {
-        if (config.getRestPort() != 0) {
-            jettyServer = new JettyServer(Strings.isBlank(config.getRestHost()) ? "0.0.0.0" : config.getRestHost(), config
-                .getRestPort(), config.getRestContextPath());
-            jettyServer.initialize();
-
-            this.registerServiceImplementation(JettyHandlerRegister.class, new JettyHandlerRegisterImpl(jettyServer));
-        } else {
-            this.receiverJettyHandlerRegister = new ReceiverJettyHandlerRegister();
-            this.registerServiceImplementation(JettyHandlerRegister.class, receiverJettyHandlerRegister);
+        JettyServerConfig jettyServerConfig = JettyServerConfig.builder()
+                                                               .host(config.getHost()).port(config.getPort())
+                                                               .contextPath(config.getContextPath())
+                                                               .jettyMinThreads(config.getJettyMinThreads())
+                                                               .jettyMaxThreads(config.getJettyMaxThreads())
+                                                               .jettyAcceptQueueSize(config.getJettyAcceptQueueSize())
+                                                               .jettyAcceptorPriorityDelta(config.getJettyAcceptorPriorityDelta())
+                                                               .jettyIdleTimeOut(config.getJettyIdleTimeOut()).build();
+
+        if (config.getPort() != 0) {
+            jettyServerConfig.setHost(Strings.isBlank(config.getHost()) ? "0.0.0.0" : config.getHost());
+            jettyServerConfig.setPort(config.getPort());
+            jettyServerConfig.setContextPath(config.getContextPath());
         }
+        jettyServer = new JettyServer(jettyServerConfig);
+        jettyServer.initialize();
```

Comment:
Can you explain more about this, the original logic is initializing `JettyServer` when `config.getRestPort() != 0`, but after your changes, `JettyServer` is always initialized, is it expected?

Reply:
The logic should not be changed. Once port is invalid, the jetty server should not be activated.